src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -940,6 +1041,9 @@ impl Tree {
 
         let [child_idx] = node.children[..] else { return None };
 
+        if node.is_exposed {
+            return None;
+        }
```

Comment:
Why does this make sense? This is definitely non-trivial, therefore needs a comment.